src/main/test/CocktailControllerTest.java

```diff
@@ -0,0 +1,138 @@
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.Main;
+import de.cocktail.controller.CocktailController;
+import de.cocktail.model.Cocktail;
+import de.cocktail.model.Ingredient;
+import de.cocktail.model.Photo;
+import de.cocktail.model.User;
+import de.cocktail.repository.CocktailRepository;
+import de.cocktail.service.CocktailService;
+import de.cocktail.web.CocktailWebOutput;
+import de.cocktail.web.IngredientWeb;
+import de.cocktail.web.UserWeb;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@AutoConfigureMockMvc
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = Main.class)
+
+public class CocktailControllerTest {
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @MockBean
+    CocktailRepository cocktailRepository;
+
+    @MockBean
```

Comment:
Our Layered architecture is as following: Controller -> Service -> Repository. If we want to test a controller, we need to mock the Service, not the Repository. Otherwise your unit tests for controller also include testing the logic of the service, which is not what we need in unit testing. Service has its own tests for its own logic and tested in isolation. If we test controller, then it is mocked. Therefore, all tests should be changed so that CocktailService is mocked. See `findCocktailsByIngredientsTitleTest()`